lib/chef/provider/git.rb

```diff
@@ -41,6 +41,20 @@ def load_current_resource
       end
 
       def define_resource_requirements
+        unless new_resource.user.nil?
+          requirements.assert(:all_actions) do |a|
+            a.assertion {
+              begin
+                require "etc" unless defined?(Etc)
```

Comment:
This require should probably just move into the method since we're calling it here, but not below which works, but is a bit confusing